hs/src/Reach/Connector/ETH_Solidity.hs

```diff
@@ -1367,7 +1368,8 @@ apiDef who ApiInfo {..} = do
   ret_ty' <- solType_ ai_ret_ty
   let ret_ty'' = ret_ty' <+> withArgLoc ai_ret_ty
   let ret = "external payable returns" <+> parens ret_ty''
-  return $ solFunction (pretty who_s) argDefns ret body
+  let mk w = solFunction (pretty w) argDefns ret body
+  return $ vsep $ mk who_s : maybe [] ((: []) . mk . bunpack) ai_alias
```

Comment:
It looks to me like on ETH you are not giving access to the Reach name AND the alias like you did on ALGO. You should decide if this is what we want and add that to the example

Reply:
It's generating both the reach name and alias; it's calling `mk` twice, one for the og name, one for the alias, if it exists.

Reply:
Ah, gross, this is duplicating the code then? Do we know the code is small?

Reply:
It's pretty small:
```
function mul2(uint256 _a0, uint256 _a1)  external payable returns (uint256 ) {
    ApiRng memory _r;
    T8 memory _t;
    _t.msg = T7(T6(_a0, _a1));
    _reach_m2(_t, _r);
    return _r.mul2;
    }
  function mul(uint256 _a0, uint256 _a1)  external payable returns (uint256 ) {
    ApiRng memory _r;
    T8 memory _t;
    _t.msg = T7(T6(_a0, _a1));
    _reach_m2(_t, _r);
    return _r.mul2;
    }
```

but I could make it like:
```
function mul(uint256 _a0, uint256 _a1)  external payable returns (uint256 ) {
    return this.mul2(_a0, _a1);
    }
```

Reply:
I think that would be better

Reply:
Sorry :(